specifyweb/frontend/js_src/lib/components/Merging/index.tsx

```diff
@@ -251,6 +257,51 @@ function Merging({
   );
 }
 
+function MergeButton<SCHEMA extends AnySchema>({
+  id,
+  mergeResource,
+}: {
+  readonly id: (suffix: string) => string;
+  readonly mergeResource: SpecifyResource<SCHEMA>;
+}): JSX.Element {
+  const [formId] = React.useState(id('form'));
+  const [saveBlocked, setSaveBlocked] = React.useState(false);
+  const [showSaveBlockedDialog, setShowBlockedDialog] = React.useState(false);
+
+  React.useEffect(() => {
+    setSaveBlocked(false);
+    return resourceOn(
+      mergeResource,
+      'blockersChanged',
+      (): void => {
+        const onlyDeferredBlockers = Array.from(
```

Comment:
this logic seems like something that would be commonly used. can you turn it into a helper function on the SaveBlocker class?

Reply:
on second thought, how is this different from `SaveBlocker.hasOnlyDeferredBlockers` function?